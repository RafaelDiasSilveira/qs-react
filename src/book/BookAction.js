// import uuid from 'uuid';

export const addBook = book => ({
  type: 'ADD_BOOK',
  // id: uuid(),
  book,
});

export default addBook;
