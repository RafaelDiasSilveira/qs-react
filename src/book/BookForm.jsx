import React from 'react';
import PropTypes from 'prop-types';

class BookForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      // codigo: props.book ? props.book.codigo : '',
      titulo: props.book ? props.book.titulo : '',
      autor: props.book ? props.book.autor : '',
      // data_publicacao: props.book ? props.book.data_publicacao : '',
      // genero: props.book ? props.book.genero : '',
    };
  }

  onTituloChange = (e) => {
    const titulo = e.target.value;
    this.setState(() => ({ titulo }));
  };

  onAutorChange = (e) => {
    const autor = e.target.value;
    this.setState(() => ({ autor }));
  };

  onSubmit = () => {

  };

  render() {
    return (
      <div>
        <form onSubmit={this.onSubmit}>

          <input
            type="text"
            placeholder="Titulo"
            value={this.state.titulo}
            onChange={this.onTituloChange}
          />

          <input
            type="text"
            placeholder="Autor"
            value={this.state.autor}
            onChange={this.onAutorChange}
          />

        </form>
      </div>
    );
  }
}

BookForm.propTypes = {
  book: PropTypes.shape({
    titulo: PropTypes.string,
    autor: PropTypes.string,
  }),
};

BookForm.defaultProps = {
  book: undefined,
};

export default BookForm;
