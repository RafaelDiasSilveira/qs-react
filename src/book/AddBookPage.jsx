import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { addBook } from './BookAction';
import BookForm from './BookForm';

export class AddBookPage extends React.Component {
  addBook = (book) => {
    this.props.onSubmit(book);
    this.props.history.push('/');
  }

  render() {
    return (
      <div>
        <h2 className="text-center">Adicionar livro</h2>
        <BookForm
          onSubmit={this.addBook}
        />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  onSubmit: book => dispatch(addBook(book)),
});

AddBookPage.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
};

export default connect(undefined, mapDispatchToProps)(AddBookPage);
