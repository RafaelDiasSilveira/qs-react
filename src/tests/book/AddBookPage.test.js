import React from 'react';
import { shallow } from "enzyme";

import { AddBookPage } from "../../book/AddBookPage";
import { books } from "../fixtures/fixtures";

let addBookSpy, historySpy, wrapper;

beforeEach(() => {
  addBookSpy = jest.fn();
  historySpy = { push: jest.fn() };
  wrapper = shallow(
    <AddBookPage
      onSubmit={addBookSpy}
      history={historySpy}
    />
  );
});

test('should render AddBookPage correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle onSubmit', () => {
  wrapper.find('BookForm').prop('onSubmit')(books[1]);

  expect(historySpy.push).toHaveBeenLastCalledWith('/');
  expect(addBookSpy).toHaveBeenLastCalledWith(books[1]);
});
