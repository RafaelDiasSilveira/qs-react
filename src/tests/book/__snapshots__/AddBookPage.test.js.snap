// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddBookPage correctly 1`] = `
<div>
  <h2
    className="text-center"
  >
    Adicionar livro
  </h2>
  <BookForm
    onSubmit={[Function]}
  />
</div>
`;
